query/executor.go

```diff
@@ -242,6 +257,8 @@ type Executor struct {
 
 	// expvar-based stats.
 	stats *Statistics
+
+	mu sync.Mutex
```

Comment:
I would not put `mu` at the bottom of the struct.  A developer going into this code might miss `mu` and not lock appropriately.

My preference is to put `mu` above all the members that it will protect, along with a comment that `mu protects all following members`. Members that don't require protection can go above `mu`.